scripts/utils/internet_detect_tool.py

```diff
@@ -0,0 +1,246 @@
+from PyQt5.QtWidgets import QApplication, QMenu
+from PyQt5.QtCore import QTimer, QThread, pyqtSignal
+import threading
+import requests
+import urllib3
+import win32api
+import win32gui
+import win32con
+import winreg
+import struct
+import signal
+import ctypes
+import time
+import os
+
+
+# Define constants
+CHECK_INTERVAL = 2  # Seconds
+CONNECT_TEST_URL = "https://www.msftconnecttest.com/connecttest.txt"
+EXPECTED_RESPONSE = "Microsoft Connect Test"
+SPI_SETDESKWALLPAPER = 20
+SPIF_UPDATEINIFILE = 0x01
+SPIF_SENDWININICHANGE = 0x02
+COLOR_DESKTOP = 1
+ICON_RED_X = "red_circle.ico"
+ICON_FLARE = "flare.ico"
+DEFAULT_BACKGROUND = os.path.join(os.environ.get('VM_COMMON_DIR'), "background.png")
+INTERNET_BACKGROUND = os.path.join(os.environ.get('VM_COMMON_DIR'),"background-internet.png")
+
+# Global variables
+app = QApplication([])
+tray_icon = None
+stop_event = threading.Event()  # To signal the background thread to stop
+hwnd = None     # We'll assign the window handle here later
+# Win32 API icon handles
+hicon_flare = None
+hicon_red_x = None
+
+def signal_handler(sig, frame):
+    print("Ctrl+C detected. Resetting background and exiting...")
+    reset_background()
```

Comment:
I know this was my idea, but I am thinking now it may be a bad idea:  If internet is enabled and you stop the tool, the internet is still enabled. :thinking: